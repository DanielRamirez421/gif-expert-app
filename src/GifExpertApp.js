import React, { useState } from 'react';

export default function GifExpertApp() {
    
    // const categories = ['OnePunch', 'SamuariX', 'DragonBall']
    const [categories, setCategories] = useState(['OnePunch', 'SamuariX', 'DragonBall'])
    const handleAdd = () => {

        // setCategories( [ ...categories, 'new Element' ] )
        setCategories( cats => [ ...cats, 'new Element' ] )
        
    }
    
    return (
      <div>
        <h2> GifExpertApp </h2>
        <hr />

        <button onClick={ handleAdd } >AddElement</button>

        <ol>
            { 
                categories.map( 
                    ( category ) => <li key={ category }>{ category }</li>
                ) 
            }
        </ol>

      </div>
    )
  }
  