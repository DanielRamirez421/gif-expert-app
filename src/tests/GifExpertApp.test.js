import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('GifExpertApp component tests', () => {
  
    test('should create GifExpertApp component', () => {
        const wrapper = shallow( <GifExpertApp /> );
        expect(wrapper).toMatchSnapshot();
    });

    test('should first', () => {
        const categories = ['Dragon Ball', 'One Punch'];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories }/> );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
    

});
