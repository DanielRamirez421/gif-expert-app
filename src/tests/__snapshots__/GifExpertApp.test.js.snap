// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp component tests should create GifExpertApp component 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</div>
`;

exports[`GifExpertApp component tests should first 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="Rick and Morty"
    key="Rick and Morty"
  />
  <GifGrid
    category="One Punch"
    key="One Punch"
  />
</div>
`;
